File: src/types/importExport.ts
// 导入导出类型

export interface ExportResult {
  success: boolean;
  rowsExported: number;
  filePath: string;
  durationMs: number;
  error?: string;
}

export interface ImportResult {
  success: boolean;
  rowsImported: number;
  durationMs: number;
  error?: string;
}

// 统一格式（V2 从 useTauri.ts 迁入）
export type ExportFormat = 'csv' | 'txt' | 'json' | 'html' | 'xml' | 'sql' | 'jsonl' | 'xlsx';
export type ImportFormat = 'csv' | 'txt' | 'json' | 'xml' | 'sql';

File: src-tauri/src/models/import_export.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ExportResult {
    pub success: bool,
    #[serde(rename = "rowsExported")]
    pub rows_exported: u64,
    #[serde(rename = "filePath")]
    pub file_path: String,
    #[serde(rename = "durationMs")]
    pub duration_ms: u64,
    pub error: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ImportResult {
    pub success: bool,
    #[serde(rename = "rowsImported")]
    pub rows_imported: u64,
    #[serde(rename = "durationMs")]
    pub duration_ms: u64,
    pub error: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CsvExportInfo {
    #[serde(rename = "filePath")]
    pub file_path: String,
    #[serde(rename = "rowsExported")]
    pub rows_exported: u64,
}
